<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="generator" content="pandoc">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
  <link rel="icon" href="../favicon.ico" />
  <link rel="stylesheet" href="../style.css">
  <title>デバイスのアップグレード後にログインできない | Techioz</title>
  <style type="text/css">code{white-space: pre;}</style>
  <script async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js?client=ca-pub-1637770322917058"
     crossorigin="anonymous"></script>
  <!-- Google tag (gtag.js) -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=G-GV53GJEY2E"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'G-GV53GJEY2E');
  </script>
<script type="application/ld+json">
{
  "@context": "http://schema.org",
  "@type": "Article",
  "headline": "デバイスのアップグレード後にログインできない",
  "author": {
    "@type": "Person",
    "address": "",
    "email": null,
    "identifier": "masyumaroking",
    "name": "masyumaroking"
  },
  "publisher": {
    "@type": "Person",
    "name": "masyumaroking"
  }
}
</script>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
  <header>
    <div class="container">
        <h1>Techioz Blog</h1>
        <nav>
            <ul>
                <li><a href="#">Home</a></li>
                <li><a href="#">Articles</a></li>
                <li><a href="#">About</a></li>
                <li><a href="#">Contact</a></li>
            </ul>
        </nav>
    </div>
  </header>
  <main>
    <div class="container">
        <article>
                      <h2>デバイスのアップグレード後にログインできない</h2>
                    <h2 id="概要">概要</h2>
                    <p>Device gem を 2.1.2 から 2.1.3
                    にアップグレードしてみました。
                    バージョンアップしたらログインできなくなりました。
                    ログには 401
                    未承認メッセージのみが表示され、例外スタック
                    トレースは表示されませんでした。</p>
                    <p>これは私が使用するログインフォームです:</p>
                    <pre><code>= form_for(resource, as: resource_name, url: session_path(resource_name)) do |f|
  .mail
    = f.label :email
    = f.email_field :email

  .pwd
    = f.label :password
    = f.password_field :password

  - if devise_mapping.rememberable?
    .remember
      = f.check_box :remember_me
      = f.label :remember_me

  .submit
    = f.submit t(&#39;login.submit&#39;)
</code></pre>
                    <p>他のバージョンにアップグレードしてみましたが、結果は同じでした。
                    また、application_controllerからprotect_from_forgeryを削除しても役に立ちませんでした。
                    現在、Rails 3.2.12 と Ruby 1.9.3-p194
                    を使用しています。</p>
                    <p>誰かアイデアはありますか？</p>
                    <h2 id="解決策">解決策</h2>
                    <p>私もDeviseで同様の経験をしました。通常、それは、暗号化アルゴリズムを変更する工夫に帰着します。</p>
                    <p>アプリケーションまたはデバイスの構成ファイルで選択した暗号化タイプを確認してください。
                    (ここでは bcrypt または sha が通常の容疑者です)</p>
        </article>
    </div>
  </main>
  <footer>
    <div class="container">
        <p>&copy; 2024 Techioz. All rights reserved.</p>
    </div>
  </footer>
</body>
</html>